src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -393,6 +393,29 @@ define('diagramStore', ['jquery', 'draw.io', 'xwiki-events-bridge'], function($)
   // We need to update the form fields before the form is validated (for Preview, Save and Save &amp; Continue).
   $(document).on('xwiki:actions:beforePreview xwiki:actions:beforeSave', updateFormFields);
 
+  var saveAttachment = function(file) {
+    file.getUi().exportToCanvas(function(canvas) {
+      if(canvas) {
+        canvas.toBlob(function(blob) {
+          let doc = new XWiki.Document(xm.documentReference);
+          let url = doc.getURL('upload');
+          let formData = new FormData();
+          formData.append('filepath', blob);
+          formData.append('filename', xm.documentReference.name + '.png');
+          formData.append('form_token', xm.form_token);
+          $.post({
+            url: url,
+            data: formData,
+            processData: false,
+            contentType: false
+          });
+        });
+      }
+    }, null, null, null, function(e) {
+      new XWiki.widgets.Notification("$services.localization.render('diagram.page.error')", 'error');
+    }, null, true, null, null, null, null, null, null, null);
```

Comment:
Do you really need to pass all these ``null`` values? I think you can stop after the ``true`` argument (the rest will be ``undefined`` if not specified which should be handled as ``null``).